src/lib/AudioUltra/Regions/Region.ts

```diff
@@ -32,6 +32,57 @@ export class Region extends Segment {
     return true;
   }
 
+  get options() {
+    return {
+      ...super.options,
+      labels: this.labels,
+      color: this.color.toString(),
+    };
+  }
+
+  renderLabels(): void {
+    if (this.labels?.length && this.controller.showLabels) {
+      const layer = this.controller.layerGroup;
+      const color = this.color;
+      const timelineTop = this.timelinePlacement;
+      const timelineLayer = this.visualizer.getLayer('timeline');
+      const timelineHeight = this.timelineHeight;
+      const top = timelineLayer?.isVisible && timelineTop ? timelineHeight : 0;
+      const labelMeasures = this.labels.map((label) => layer.context.measureText(label));
+
+      const height = labelMeasures.reduce((accumulator, currentValue) => {
```

Comment:
How does this work with multiple labels? I forgot to check that case, but is this stacking the labels or taking the average height of all the text heights? Maybe add a comment about the approach used here.

Reply:
It stacks them, so the reduce calculates the combined height of the box and the heightCalc in the labels for each determines the vertical position of the text